mreg_cli/cli.py

```diff
@@ -44,6 +44,16 @@
     # Can't use _SubParsersAction as generic in Python <3.9
     SubparserType = argparse._SubParsersAction[argparse.ArgumentParser]  # type: ignore # private attribute
 
+    # Subclasses of BaseCommand change the constructor signature from
+    # 6 arguments to a single argument `cli`. That means we cannot
+    # declare the list of commands as a list of `type[BaseCommand]`
+    # and then instantiate them. Instead, we need to declare this interface
+    # which lets the type checker understand what kind of classes we are
+    # trying to instantiate.
+    class BaseCommandSubclass(Protocol):  # noqa: D101
```

Comment:
Should we do something like:

`class BaseCommandSubclass(Protocol):  # noqa: D101 (undocumented-public-class)` 

So we make it clear what we're disabling? I haven't memorised *all* the ruff codes yet. :)

Reply:
Sure, we can do that.